### PR TITLE
[Merged by Bors] - feat: in finite dimensions, if a linear endomorphism is triangularizable, so is its restriction to any invariant submodule

### PR DESCRIPTION
**Pull request by ocfnash**
---

Split out from #7963

- [x] depends on: #8177

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by ocfnash** [comment]
I note the following request to reviewers from @semorrison on Zulip:
```quote
if you see a PR that is renaming a file, but git does not report it as a rename (because too much stuff was changed at the same time), can we please ask the author to make separate PRs doing the rename and making the changes?

Renames that are not detected as renames are very likely to break nightly-testing. (Thanks to Matthew Ballard for salvaging one such breakage overnight.)
```
Thus if the proposed rename in this PR is accepted, we should not merge this PR directly but split out a split rename-only PR first. I will wait for review before I make such a split.

**Comment by jcommelin** [comment]
I would like to see the renaming PR (-;

**Comment by ocfnash** [comment]
> I would like to see the renaming PR (-;

Created here #8318

File: Mathlib/LinearAlgebra/Eigenspace/Triangularizable.lean [hunk 1]
```diff
@@ -13,29 +13,35 @@ import Mathlib.FieldTheory.IsAlgClosed.Spectrum
 
 This file contains basic results relevant to the triangularizability of linear endomorphisms.
 
-* Every linear operator on a vector space over an algebraically closed field has an eigenvalue.
-* The generalized eigenvectors span the entire vector space.
+## Main definitions / results
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Eigenspace/Triangularizable.lean:16]: I think `exists_eigenvalue` can also be added as item to this list. What do you think?

File: Mathlib/LinearAlgebra/Eigenspace/Triangularizable.lean [hunk 2]
```diff
@@ -13,29 +13,35 @@ import Mathlib.FieldTheory.IsAlgClosed.Spectrum
...
+
+ * `Module.End.iSup_generalizedEigenspace_eq_top`: in finite dimensions, over an algebraically
+   closed field, the generalized eigenspaces of any linear endomorphism span the whole space.
+ * `Module.End.iSup_generalizedEigenspace_restrict_eq_top`: in finite dimensions, if the
+   generalized eigenspaces of a linear endomorphism span the whole space then the same is true of
+   its restriction to any invariant submodule.
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Eigenspace/Triangularizable.lean:22]: Please add these descriptions as docstrings to these results. I understand the names after staring at the type for 5 secs. I guess these natural language docstrings might speed up that process.

**Review by jcommelin — APPROVED** [review]
bors d+

File: Mathlib/LinearAlgebra/Eigenspace/Triangularizable.lean (hunk 1 above)
Comment (ocfnash) [reply Mathlib/LinearAlgebra/Eigenspace/Triangularizable.lean:16]: Agreed; done.

File: Mathlib/LinearAlgebra/Eigenspace/Triangularizable.lean (hunk 2 above)
Comment (ocfnash) [reply Mathlib/LinearAlgebra/Eigenspace/Triangularizable.lean:22]: Good point; done.

**Pull request closed without merging**